devdocs/troubleshooting-guide.md

```diff
@@ -0,0 +1,198 @@
+
+    ![setup-checklist-4.png](images/setup-checklist-4.png)
+
+6. **Check for compilation errors**
+
+    Markers tab should have 0 errors.
+
+    ![setup-checklist-5.png](images/setup-checklist-5.png)
+
+    If it shows errors, try refreshing the project (right-click on the project in Project explorer) followed by a Project → Clean.
+
+### Common errors and solutions
+
+1. **ERROR**: Eclipse complains about FileWriter and ConsoleHandler e.g. java.io.FileWriter is not supported by Google App Engine's Java runtime environment.
+
+  **SOLUTION**: These are OK because the offending files belong to test driver and will not be deployed to GAE. Choose to 'Quick Fix' and then choose to exclude those files from validation.
+
+2. **ERROR**: Eclipse complains "...your project must be configured to use a jdk in order to use jsp".
+
+  **SOLUTION**: This happens because Eclipse is only aware of JRE, not JDK (Compiling JSP requires the JDK). Go to Window → Preferences → Java → Installed JREs. You will note that a JRE path is the one selected, not a JDK path. To fix this, Click add → Standard VM, then for the JRE Path enter the path of the jre folder inside your jdk installation folder. e.g., C:\jdk1.7\jre Now you should see all of the JARs added to the library section.
+
+3. **ERROR**: When trying to deploy, Eclipse complains "... Cannot get the System Java Compiler. Please use a JDK, not a JRE.".
+
+  **SOLUTION**: You can force Eclipse to use the JDK (instead of JRE) by modifying the eclipse.ini file. See here for more details.
+
+4. **ERROR (on Windows)**: Dev server launched by Eclipse keeps running even after closing Eclipse. After restarting Eclipse, you'll be able to relaunch dev server on the same port but requests will be received by the previous server instance.
+
+  **SOLUTION**: If Eclipse crashes while development server is running inside Eclipse, the server might keep running even after Eclipse is closed. Go to Windows Task Manager and kill processes named javaw.exe.
+
+5. **ERROR**: Eclipse complains "file out of sync".
+
+  **SOLUTION**: 'refresh' the project in eclipse.
+
+6. **ERROR**: Eclipse complains "There are no JREs installed in the workplace that are strictly compatible with this environment.".
+
+  **SOLUTION**: Eclipse may be using an incompatible version of the JRE Library (with respect to TEAMMATES) for the current JRE definition. System Library for JRE should be set to the workspace default, after appropriate JRE definition has been added (covered in existing point#2 of the troubleshooting section).
+  
+   
+   → Right-click on project → Properties → Java Build Path → "Libraries" tab → Select JRE System Library from the list → Edit… → Select and mark radio button for "Workspace default JRE"
+   
+
+7. **ERROR**: Everything seems normal and the dev server starts normally. But you get a 'Class not found" or "Class cannot initialize" error when you try to access any non-static pages of the app.
```

Comment:
The quotes used in "Class not found" are not consistent. Use double quotes for consistency.